osgp/platform/osgp-adapter-kafka-distributionautomation/src/test/java/org/opensmartgridplatform/adapter/kafka/da/application/mapping/LowVoltageMessageFactory.java

```diff
@@ -0,0 +1,115 @@
+import java.util.ArrayList;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LowVoltageMessageFactory extends MessageFactory {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(LowVoltageMessageFactory.class);
+
+  public enum Version {
+    VERSION_1,
+    VERSION_2;
+  }
+
+  public static List<Analog> expectedMeasurements(final Version version) {
+    if (Version.VERSION_1 == version) {
+      return expectedMeasurementsVersion1();
+    } else if (Version.VERSION_2 == version) {
+      return expectedMeasurementsVersion2();
+    } else {
+      LOGGER.error("Not given the an expected version. Given version: {}", version);
+      return null;
+    }
+  }
+
+  public static List<Analog> expectedMetaMeasurements() {
+    final List<Analog> measurements = new ArrayList<>();
+    measurements.add(createAnalog("Frequency", 49.98f, UnitSymbol.Hz));
+    measurements.add(createAnalog("Temperature", 12.0f, UnitSymbol.degC));
+    measurements.add(createAnalog("IRMS-N", 0.11f, UnitSymbol.A));
+    return measurements;
+  }
+
+  private static List<Analog> expectedMeasurementsVersion1() {
+    final List<Analog> measurements = new ArrayList<>();
+    measurements.add(createAnalog("U-L1", 0.1f, UnitSymbol.V));
+    measurements.add(createAnalog("U-L2", 0.2f, UnitSymbol.V));
+    measurements.add(createAnalog("U-L3", 0.3f, UnitSymbol.V));
+    generalExpectedMeasurements(measurements, 0.4);
+    return measurements;
+  }
+
+  private static List<Analog> expectedMeasurementsVersion2() {
+    final List<Analog> measurements = new ArrayList<>();
+    measurements.add(createAnalog("U-avg", 0.1f, UnitSymbol.V));
+    generalExpectedMeasurements(measurements, 0.2);
+    double value = measurements.size() * 0.1;
+    measurements.add(createAnalog("IrmsN", (float) (value = value + 0.1), UnitSymbol.A));
+    measurements.add(createAnalog("Pp", (float) (value = value + 0.1), UnitSymbol.none));
+    measurements.add(createAnalog("Pm", (float) (value = value + 0.1), UnitSymbol.none));
+    measurements.add(createAnalog("Qp", (float) (value = value + 0.1), UnitSymbol.none));
+    measurements.add(createAnalog("Qm", (float) (value = value + 0.1), UnitSymbol.none));
+    measurements.add(createAnalog("U-L1", (float) (value = value + 0.1), UnitSymbol.V));
+    measurements.add(createAnalog("U-L2", (float) (value = value + 0.1), UnitSymbol.V));
+    measurements.add(createAnalog("U-L3", (float) (value = value + 0.1), UnitSymbol.V));
+    measurements.add(createAnalog("Temp", (float) (value = value + 0.1), UnitSymbol.C));
+    measurements.add(createAnalog("F", (float) (value = value + 0.1), UnitSymbol.Hz));
+    return measurements;
+  }
+
+  private static void generalExpectedMeasurements(final List<Analog> measurements, double d) {
+    measurements.add(createAnalog("I-L1", (float) d, UnitSymbol.A));
+    measurements.add(createAnalog("I-L2", (float) (d = d + 0.1), UnitSymbol.A));
```

Comment:
When using a float parameter the cast is no longer needed. Furthermore, for these inline updated values I think it would be okay to use the shorter `value += 0.1F`.
Then it will become something like:
```suggestion
    measurements.add(createAnalog("I-L2", value += 0.1F, UnitSymbol.A));
```

Reply:
Sadly this is not possible, because this adds random numbers add the end of the value. See stacktrace below. I would like to get the a float working, but I don't see how. I tried round it of but couldn't find a good solution.
```
org.opentest4j.AssertionFailedError: 
Expecting:
 <[{"AnalogValues": [{"MeasurementValueQuality": null, "value": 0.1}], "accumulation": "none", "description": "U-L1", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "V"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 0.2}], "accumulation": "none", "description": "U-L2", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "V"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 0.3}], "accumulation": "none", "description": "U-L3", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "V"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 0.4}], "accumulation": "none", "description": "I-L1", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 0.5}], "accumulation": "none", "description": "I-L2", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 0.6}], "accumulation": "none", "description": "I-L3", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 0.7}], "accumulation": "none", "description": "Tot-P", "measuringPeriod": "none", "unitMultiplier": "k", "unitSymbol": "W"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 0.8}], "accumulation": "none", "description": "Tot-Q", "measuringPeriod": "none", "unitMultiplier": "k", "unitSymbol": "VAr"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 0.9}], "accumulation": "none", "description": "P-L1", "measuringPeriod": "none", "unitMultiplier": "k", "unitSymbol": "W"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 1.0}], "accumulation": "none", "description": "P-L2", "measuringPeriod": "none", "unitMultiplier": "k", "unitSymbol": "W"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 1.1}], "accumulation": "none", "description": "P-L3", "measuringPeriod": "none", "unitMultiplier": "k", "unitSymbol": "W"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 1.2}], "accumulation": "none", "description": "Q-L1", "measuringPeriod": "none", "unitMultiplier": "k", "unitSymbol": "VAr"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 1.3}], "accumulation": "none", "description": "Q-L2", "measuringPeriod": "none", "unitMultiplier": "k", "unitSymbol": "VAr"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 1.4}], "accumulation": "none", "description": "Q-L3", "measuringPeriod": "none", "unitMultiplier": "k", "unitSymbol": "VAr"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 1.5}], "accumulation": "none", "description": "PF-L1", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "none"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 1.6}], "accumulation": "none", "description": "PF-L2", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "none"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 1.7}], "accumulation": "none", "description": "PF-L3", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "none"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 1.8}], "accumulation": "none", "description": "THDi-L1", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "PerCent"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 1.9}], "accumulation": "none", "description": "THDi-L2", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "PerCent"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 2.0}], "accumulation": "none", "description": "THDi-L3", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "PerCent"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 2.1}], "accumulation": "none", "description": "H3-I1", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 2.2}], "accumulation": "none", "description": "H3-I2", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 2.3}], "accumulation": "none", "description": "H3-I3", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 2.4}], "accumulation": "none", "description": "H5-I1", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 2.5}], "accumulation": "none", "description": "H5-I2", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 2.6}], "accumulation": "none", "description": "H5-I3", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 2.7}], "accumulation": "none", "description": "H7-I1", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 2.8}], "accumulation": "none", "description": "H7-I2", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 2.9}], "accumulation": "none", "description": "H7-I3", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 3.0}], "accumulation": "none", "description": "H9-I1", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 3.1}], "accumulation": "none", "description": "H9-I2", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 3.2}], "accumulation": "none", "description": "H9-I3", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 3.3}], "accumulation": "none", "description": "H11-I1", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 3.4}], "accumulation": "none", "description": "H11-I2", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 3.5}], "accumulation": "none", "description": "H11-I3", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 3.6}], "accumulation": "none", "description": "H13-I1", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 3.7}], "accumulation": "none", "description": "H13-I2", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 3.8}], "accumulation": "none", "description": "H13-I3", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 3.9}], "accumulation": "none", "description": "H15-I1", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 4.0}], "accumulation": "none", "description": "H15-I2", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 4.1}], "accumulation": "none", "description": "H15-I3", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"}]>
to be equal to:
 <[{"AnalogValues": [{"MeasurementValueQuality": null, "value": 0.1}], "accumulation": "none", "description": "U-L1", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "V"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 0.2}], "accumulation": "none", "description": "U-L2", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "V"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 0.3}], "accumulation": "none", "description": "U-L3", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "V"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 0.4}], "accumulation": "none", "description": "I-L1", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 0.5}], "accumulation": "none", "description": "I-L2", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 0.6}], "accumulation": "none", "description": "I-L3", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 0.70000005}], "accumulation": "none", "description": "Tot-P", "measuringPeriod": "none", "unitMultiplier": "k", "unitSymbol": "W"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 0.8000001}], "accumulation": "none", "description": "Tot-Q", "measuringPeriod": "none", "unitMultiplier": "k", "unitSymbol": "VAr"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 0.9000001}], "accumulation": "none", "description": "P-L1", "measuringPeriod": "none", "unitMultiplier": "k", "unitSymbol": "W"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 1.0000001}], "accumulation": "none", "description": "P-L2", "measuringPeriod": "none", "unitMultiplier": "k", "unitSymbol": "W"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 1.1000001}], "accumulation": "none", "description": "P-L3", "measuringPeriod": "none", "unitMultiplier": "k", "unitSymbol": "W"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 1.2000002}], "accumulation": "none", "description": "Q-L1", "measuringPeriod": "none", "unitMultiplier": "k", "unitSymbol": "VAr"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 1.3000002}], "accumulation": "none", "description": "Q-L2", "measuringPeriod": "none", "unitMultiplier": "k", "unitSymbol": "VAr"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 1.4000002}], "accumulation": "none", "description": "Q-L3", "measuringPeriod": "none", "unitMultiplier": "k", "unitSymbol": "VAr"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 1.5000002}], "accumulation": "none", "description": "PF-L1", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "none"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 1.6000003}], "accumulation": "none", "description": "PF-L2", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "none"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 1.7000003}], "accumulation": "none", "description": "PF-L3", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "none"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 1.8000003}], "accumulation": "none", "description": "THDi-L1", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "PerCent"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 1.9000003}], "accumulation": "none", "description": "THDi-L2", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "PerCent"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 2.0000002}], "accumulation": "none", "description": "THDi-L3", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "PerCent"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 2.1000001}], "accumulation": "none", "description": "H3-I1", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 2.2}], "accumulation": "none", "description": "H3-I2", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 2.3}], "accumulation": "none", "description": "H3-I3", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 2.3999999}], "accumulation": "none", "description": "H5-I1", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 2.4999998}], "accumulation": "none", "description": "H5-I2", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 2.5999997}], "accumulation": "none", "description": "H5-I3", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 2.6999996}], "accumulation": "none", "description": "H7-I1", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 2.7999995}], "accumulation": "none", "description": "H7-I2", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 2.8999994}], "accumulation": "none", "description": "H7-I3", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 2.9999993}], "accumulation": "none", "description": "H9-I1", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 3.0999992}], "accumulation": "none", "description": "H9-I2", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 3.199999}], "accumulation": "none", "description": "H9-I3", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 3.299999}], "accumulation": "none", "description": "H11-I1", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 3.399999}], "accumulation": "none", "description": "H11-I2", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 3.4999988}], "accumulation": "none", "description": "H11-I3", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 3.5999987}], "accumulation": "none", "description": "H13-I1", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 3.6999986}], "accumulation": "none", "description": "H13-I2", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 3.7999985}], "accumulation": "none", "description": "H13-I3", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 3.8999984}], "accumulation": "none", "description": "H15-I1", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 3.9999983}], "accumulation": "none", "description": "H15-I2", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"},
    {"AnalogValues": [{"MeasurementValueQuality": null, "value": 4.0999985}], "accumulation": "none", "description": "H15-I3", "measuringPeriod": "none", "unitMultiplier": "none", "unitSymbol": "A"}]>
when comparing elements using field/property by field/property comparator on all fields/properties except ["mRID"]
Comparators used:
- for elements fields (by type): {Double -> DoubleComparator[precision=1.0E-15], Float -> FloatComparator[precision=1.0E-6]}
- for elements (by type): {Double -> DoubleComparator[precision=1.0E-15], Float -> FloatComparator[precision=1.0E-6]}
but was not.
```